sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -258,7 +256,154 @@ class FilesElement extends Polymer.Element {
+        inputValue: selectedObject.name,
+        okLabel: 'Rename',
+      };
+
+      // Only if the dialog has been confirmed with some user input, rename the
+      // selected item. Then if that is successful, and reload the file list.
+      return Utils.getUserInputAsync(inputOptions)
+        .then((closeResult: DialogCloseResult) => {
+          if (closeResult.confirmed && closeResult.userInput) {
+            let newName = this.currentPath + '/' + closeResult.userInput;
+            return ApiManager.renameItem(selectedObject.path, newName)
+              .then(() => this._fetchFileList());
+              // TODO: [yebrahim] Re-select the renamed item after refresh
+          } else {
+            return Promise.resolve(null);
+          }
+        })
+    } else {
+      // TODO: Handle rename errors properly by showing some message to the user
+      return Promise.resolve(null);
+    }
+  }
+
+  /**
+   * Creates an input modal to get the user confirmation with a list of the items
+   * to be deleted, then calls the ApiManager for each of these items to delete,
+   * then refreshes the file list.
+   */
+  _deleteSelectedItems() {
+
+    const selectedIndices = this.$.files.getSelectedIndices();
+    if (selectedIndices.length) {
+      // Build friendly title and body messages that adapt to the number of items.
+      const num = selectedIndices.length;
+      let title = 'Delete ';
+
+      // Title
+      if (num === 1) {
+        const i = selectedIndices[0];
+        const selectedObject = this._fileList[i];
+        title += selectedObject.type.toString();
+      } else {
+        title += num + ' items';
+      }
+
+      // Body
+      let itemList = '<ul>\n';
+      selectedIndices.forEach((fileIdx: number, i: number) => {
+        if (i < FilesElement._deleteListLimit)
+          itemList += '<li>' + this._fileList[fileIdx].name + '</li>\n';
+      });
+      if (num > FilesElement._deleteListLimit) {
+        itemList += '+ ' + (num - FilesElement._deleteListLimit) + ' more.';
+      }
+      itemList += '</ul>'
+      const bodyHtml = '<div>Are you sure you want to delete:</div>' + itemList;
+
+      // Open a dialog to let the user confirm deleting the list of selected items.
+      const inputOptions: DialogOptions = {
+        title: title,
+        bodyHtml: bodyHtml,
+        okLabel: 'Delete',
+      };
+
+      // Only if the dialog has been confirmed, call the ApiManager to delete each
+      // of the selected items, and wait for all promises to finish. Then if that
+      // is successful, reload the file list.
+      return Utils.getUserInputAsync(inputOptions)
+        .then((closeResult: DialogCloseResult) => {
+          if (closeResult.confirmed) {
+            let deletePromises = selectedIndices.map((i: number) => {
+              return ApiManager.deleteItem(this._fileList[i].path);
+            });
+            return Promise.all(deletePromises)
+              .then(() => this._fetchFileList());
```

Comment:
You mean the deletes?
No, no message is displayed for errors yet, I have a few TODO lines in code to add those later.
You're right, if at least one delete gets through and then a failure happens, fetching the file list won't happen. I'll add a comment to think about this when adding error handling.